quickwit-cli/src/lib.rs

```diff
@@ -135,6 +154,30 @@ pub struct GarbageCollectIndexArgs {
     pub dry_run: bool,
 }
 
+pub async fn create_read_split_cli(args: ReadSplitArgs) -> anyhow::Result<()> {
+    debug!(args = ?args, "read-split");
+
+    let split_file = PathBuf::from(format!("{}.split", args.split_id));
+
+    let storage_uri_resolver = quickwit_storage_uri_resolver();
+    let metastore_uri_resolver = MetastoreUriResolver::default();
+    let metastore = metastore_uri_resolver.resolve(&args.metastore_uri).await?;
+
+    let index_metadata = metastore.index_metadata(&args.index_id).await?;
+
+    let index_storage = storage_uri_resolver.resolve(&index_metadata.index_uri)?;
+
+    let bundle = index_storage.get_all(split_file.as_path()).await?;
```

Comment:
Are we downloading all the whole split here just for showing some stats?

Reply:
There is a discussion about that here: https://github.com/quickwit-inc/quickwit/issues/376

Reply:
Yes, without the file len api, it's not possible otherwise

Reply:
fixed, downloading only the necessary part